app/src/main/java/ch/heigvd/pro/b04/android/Question/QuestionViewModel.java

```diff
@@ -38,49 +38,49 @@
-                                Long delta = System.currentTimeMillis() - lastRequest.getOrDefault(received, 0L);
-                                // If enough time has passed, update the answer
-                                if (delta > STABILIZATION_DELAY) {
-                                    transferred.add(received);
-                                    lastRequest.remove(received);
-                                } else {
-                                    // If not enough time has passed, use the last known state
-                                    currentAnswers.getValue().stream()
-                                            .filter(ans -> ans.equals(received))
-                                            .findFirst()
-                                            .ifPresent(transferred::add);
-                                }
-                            }
-
-                            return transferred;
-                        })
+                unit -> Rockin.api().getAnswers(currentQuestion.getValue(), token)
         );
 
-        currentAnswers.addSource(transformQuestion, answers -> currentAnswers.postValue(answers));
-        currentAnswers.addSource(transformDelayed, answers -> currentAnswers.postValue(answers));
-    }
+        answerResponse.addSource(transformQuestion, response -> answerResponse.postValue(response));
+        answerResponse.addSource(transformDelayed, response -> answerResponse.postValue(response));
+
+        currentAnswers = Transformations.switchMap(answerResponse, response -> {
+            if (! response.response().isPresent()) {
+                return new MutableLiveData<>();
+            }
+
+            List<Answer> transferred = new ArrayList<>();
+            for (Answer received : response.response().get()) {
+                Long delta = System.currentTimeMillis() - lastRequest.getOrDefault(received, 0L);
+                // If enough time has passed, update the answer
+                if (delta > STABILIZATION_DELAY) {
+                    transferred.add(received);
+                    lastRequest.remove(received);
+                } else {
+                    // If not enough time has passed, use the last known state
+                    currentAnswers.getValue().stream()
+                            .filter(ans -> ans.equals(received))
+                            .findFirst()
+                            .ifPresent(transferred::add);
+                }
+            }
+
+            return new MutableLiveData<>(transferred);
+        });
 
-    private static LiveData<List<Answer>> questionToAnswer(Question question, String token) {
-        return LiveDataUtils.ignorePendingAndErrors(Rockin.api().getAnswers(
-                question.getIdModerator(),
-                question.getIdPoll(),
-                question.getIdQuestion(),
-                token));
+        responseError = Transformations.map(answerResponse, response -> response.isFailure());
```

Comment:
Maybe we use wrap it all in a [distinctUntilChanged(..)](https://developer.android.com/reference/androidx/lifecycle/Transformations#distinctUntilChanged(androidx.lifecycle.LiveData<X>)) call.